src/test/ui/impl-trait/where-allowed-2.stderr

```diff
@@ -1,10 +1,12 @@
-error[E0720]: opaque type expands to a recursive type
+error[E0720]: cannot resolve opaque type to a concrete type
   --> $DIR/where-allowed-2.rs:6:30
    |
 LL | fn in_adt_in_return() -> Vec<impl Debug> { panic!() }
-   |                              ^^^^^^^^^^ expands to a recursive type
+   |                              ^^^^^^^^^^    -------- this returned value is of `!` type
+   |                              |
+   |                              cannot resolve to a concrete type
    |
-   = note: type resolves to itself
+   = help: this error will resolve once the item's body returns a concrete type
```

Comment:
I'm not sure "concrete" is the right terminology. I guess the "recursive type" comes from inference variables unifying?